packages/tabs/tests/integration/components/nucleus-tabs-test.js

```diff
@@ -0,0 +1,214 @@
+        {{#tabs.panel name="about" props=tabs.props }}
+          <div>This is the about section</div>
+        {{/tabs.panel}}
+      {{/nucleus-tabs}}
+    `);
+
+    await click('.nucleus-tabs .nucleus-tabs__list__item:not(.active)');
+    assert.ok(beforeChangeAction.calledOnce, 'beforeChange action has been called.');
+    assert.ok(onChangeAction.calledOnce, 'beforeChange action has been called.');
+    assert.verifySteps(['beforeChange', 'onChange']);
+  });
+
+  test('it has accessibility attributes', async function(assert) {
+    await render(sampleTabsTemplate);
+    assert.dom('.nucleus-tabs .nucleus-tabs__list').hasAttribute('role', 'tablist');
+    assert.dom('.nucleus-tabs .nucleus-tabs__list').hasAttribute('aria-label', 'site-navigation');
+    assert.dom('.nucleus-tabs .nucleus-tabs__list button').hasAttribute('role', 'tab');
+    assert.dom('.nucleus-tabs .nucleus-tabs__list button.active').hasAttribute('aria-selected', 'true');
+    assert.dom('.nucleus-tabs .nucleus-tabs__list button:not(.active)').hasAttribute('aria-selected', 'false');
+    assert.dom('.nucleus-tabs .nucleus-tabs__list button').hasAttribute('aria-controls');
+    assert.dom('.nucleus-tabs .nucleus-tabs__panel').hasAttribute('role', 'tabpanel');
+    assert.dom('.nucleus-tabs .nucleus-tabs__panel').hasAttribute('aria-labelledby');
+  });
+
+  test('it passes a11y tests', async function(assert) {
+    await render(sampleTabsTemplate);
+
+    return a11yAudit(this.element).then(() => {
+      assert.ok(true, 'no a11y errors found!');
+    });
+  });
+  
+  test('visual regression for default style tabs', async function(assert) {
+    await render(hbs`
+      {{#nucleus-tabs description="site-navigation" selected="home" as |tabs|}}
+        {{#tabs.panel name="home" props=tabs.props }}
+          <div>This is the home section</div>
+        {{/tabs.panel}}
+      {{/nucleus-tabs}}
+    `);
+    await backstop(assert, {scenario:{misMatchThreshold: 0.1}}); 
+  });
+
+  test('visual regression for background style tabs', async function(assert) {
+    await render(hbs`
+      {{#nucleus-tabs description="site-navigation" variant="background" selected="home" as |tabs|}}
+        {{#tabs.panel name="home" props=tabs.props }}
+          <div>This is the home section</div>
+        {{/tabs.panel}}
+      {{/nucleus-tabs}}
+    `);
+    await backstop(assert, {scenario:{misMatchThreshold: 0.1}}); 
+  });
+
+  test('visual regression for disabled tabs', async function(assert) {
+    await render(hbs`
+      {{#nucleus-tabs description="site-navigation" variant="background" selected="home" as |tabs|}}
+        {{#tabs.panel name="home" props=tabs.props }}
+          <div>This is the home section</div>
+        {{/tabs.panel}}
+        {{#tabs.panel name="about" props=tabs.props disabled="true" }}
+          <div>This is the home section</div>
+        {{/tabs.panel}}
+      {{/nucleus-tabs}}
+    `);
+    await backstop(assert, {scenario:{misMatchThreshold: 0.1}}); 
+  });
```

Comment:
Visual regression tests are expensive, i.e they take a long time to run and taking snapshots are quite costly too. So try to render multiple variations in one test as much as possible.

Reply:
Merged visual regression test cases into one as suggested.

Merging cases into one also led to reducing backstop 'misMatchThreshold'. Something to be wary of next time. Else, cases passing when they should fail.

Reply:
> Merging cases into one also led to reducing backstop 'misMatchThreshold'. Something to be wary of next time.

I have already brought this up in an issue - https://github.com/freshdesk/nucleus/issues/143
We are currently letting the developer be the better judge of the misMatchThreshold depending on the way they have written the test. It is not necessary that all visual regression tests should have the same misMatchThreshold. We will have to revisit the already existing components too and change threshold values if required